core/io/resource_format_binary.cpp

```diff
@@ -711,16 +707,14 @@ Error ResourceInteractiveLoaderBinary::poll() {
 	Object *obj = ClassDB::instance(t);
 	if (!obj) {
 		error = ERR_FILE_CORRUPT;
-		ERR_EXPLAIN(local_path + ":Resource of unrecognized type in file: " + t);
-		ERR_FAIL_V(ERR_FILE_CORRUPT);
+		ERR_FAIL_V_MSG(ERR_FILE_CORRUPT, local_path + ":Resource of unrecognized type in file: " + t + ".");
 	}
 
 	Resource *r = Object::cast_to<Resource>(obj);
 	if (!r) {
 		error = ERR_FILE_CORRUPT;
-		ERR_EXPLAIN(local_path + ":Resource type in resource field not a resource, type is: " + obj->get_class());
 		memdelete(obj); //bye
-		ERR_FAIL_V(ERR_FILE_CORRUPT);
+		ERR_FAIL_V_MSG(ERR_FILE_CORRUPT, local_path + ":Resource type in resource field not a resource, type is: " + obj->get_class() + ".");
```

Comment:
This is going to crash, you can dereference `obj` after deleting it. So you need to savev `obj->class()` in a temporary variable before freeing it.